### ISSUE TITLE
Object.get_property_list ignore internal class

### ISSUE THREAD

**Issue by profelis** [godotengine/godot#6308]
**Operating system or device - Godot version:**
Godot 2.1

**Issue description** (what happened, and what was expected):
get_property_list doesn't return Reference's fields

**Steps to reproduce:**

```
func _ready():
    var test = Test.new()
    print(test.get_property_list()) # [(hint:0), (hint_string:), (name:Reference), (type:0), (usage:256)]

class Test:
    var a = 10
    var b = "test"
    var c = true
```

**Comment by profelis** [comment]
fyi, overriding `_use_builtin_script` in `Reference` class resolve problem. But I'm not sure about this fix.

**Comment by bojidar-bg** [comment]
Wouldn't checking if it is either with `_use_builtin_script` on or has a script instance do it?
E.g.
https://github.com/godotengine/godot/blob/1dfa52ba59d6e6d9660e82202a77c2233b6f56e0/core/object.cpp#L535
```patch
void Object::get_property_list(List<PropertyInfo> *p_list,bool p_reversed) const {

	if (script_instance && p_reversed) {
		p_list->push_back( PropertyInfo(Variant::NIL,"Script Variables",PROPERTY_HINT_NONE,String(),PROPERTY_USAGE_CATEGORY));
		script_instance->get_property_list(p_list);
	}

	_get_property_listv(p_list,p_reversed);

-	if (!_use_builtin_script())
+	if (!_use_builtin_script() && !script_instance)
		return;

	if (!is_type("Script")) // can still be set, but this is for userfriendlyness
		p_list->push_back( PropertyInfo( Variant::OBJECT, "script/script", PROPERTY_HINT_RESOURCE_TYPE, "Script",PROPERTY_USAGE_DEFAULT|PROPERTY_USAGE_STORE_IF_NONZERO));
	if (!metadata.empty())
		p_list->push_back( PropertyInfo( Variant::DICTIONARY, "__meta__", PROPERTY_HINT_NONE,"",PROPERTY_USAGE_NOEDITOR|PROPERTY_USAGE_STORE_IF_NONZERO));
	if (script_instance && !p_reversed) {
		p_list->push_back( PropertyInfo(Variant::NIL,"Script Variables",PROPERTY_HINT_NONE,String(),PROPERTY_USAGE_CATEGORY));
		script_instance->get_property_list(p_list);
	}

}
```
(Since it would return the script properties just fine if p_reversed is on anyway)

**Comment by bojidar-bg** [comment]
After speaking with @reduz on IRC, turns out that `_use_builtin_script` would be deprecated & removed in 3.0 anyway. So, not much to do until then.

**Comment by reduz** [comment]
this was fixed in 3.0
